dpypelines/pipeline/generic_file_ingress_v1.py

```diff
@@ -0,0 +1,184 @@
+        pipeline_config (dict): Dictionary of configuration details required to run the pipeline (determined by dataset id)
+
+    Raises:
+        Exception: If any unexpected error occurs.
+    """
+    # Create notifier from webhook env var
+    try:
+        de_notifier: BasePipelineNotifier = notifier_from_env_var_webhook(
+            "DE_SLACK_WEBHOOK"
+        )
+        logger.info("Notifier created", data={"notifier": de_notifier})
+    except Exception as err:
+        logger.error("Error occurred when creating notifier", err)
+        raise err
+
+    try:
+        submitter_email = get_submitter_email()
+        logger.info(
+            "Got submitter email",
+            data={"submitter_email": submitter_email},
+        )
+    except Exception as err:
+        logger.error("Error occurred when getting submitter email", err)
+        de_notifier.failure()
+        raise err
+
+    # Create email client from env var
+    try:
+        email_client = get_email_client()
+        logger.info("Created email client", data={"email_client": email_client})
+    except Exception as err:
+        logger.error("Error occurred when creating email client", err)
+        de_notifier.failure()
+        raise err
+
+    # Get Upload Service URL from environment variable
+    try:
+        upload_url = os.environ.get("UPLOAD_SERVICE_URL", None)
+        assert (
+            upload_url is not None
+        ), "UPLOAD_SERVICE_URL environment variable not set."
+        logger.info("Got Upload Service URL", data={"upload_url": upload_url})
+    except Exception as err:
+        logger.error("Error occurred when getting Upload Service URL", err)
+        de_notifier.failure()
+        raise err
+
+    # Attempt to access the local data store
+    try:
+        local_store = LocalDirectoryStore(files_dir)
+        files_in_directory = local_store.get_file_names()
+        logger.info(
+            "Local data store created",
+            data={
+                "local_store_dir": files_dir,
+                "files_in_directory": files_in_directory,
+            },
+        )
+    except Exception as err:
+        logger.error(
+            "Error occurred when creating local data store from files directory",
+            err,
+            data={"local_store_dir": files_dir},
+        )
+        de_notifier.failure()
+        raise err
+
+    # Extract the patterns for required files from the pipeline configuration
+    try:
+        required_file_patterns = get_required_files_patterns(pipeline_config)
+        logger.info(
+            "Required file patterns retrieved from pipeline config",
+            data={"required_file_patterns": required_file_patterns},
+        )
+    except Exception as err:
+        files_in_directory = local_store.get_file_names()
+        logger.error(
+            "Error occurred when getting required file patterns",
+            err,
+            data={
+                "files_in_directory": files_in_directory,
```

Comment:
you don't need this information. There's an error getting the required file patterns from the config. The. files that happen to be in the directory aren't relevant to that and might confuse people.

could you also remove that data field from dataset_ingress_v1 please.

Reply:
Thanks.
`"files_in_directory": files_in_directory` removed from the data field.
The data field is kept in the code with the name-value `"pipeline_config": pipeline_config` incase we need to see error details in the config file